client.go

```diff
@@ -455,10 +412,9 @@ func retry(ctx context.Context, f func() (interface{}, bool, error)) (interface{
 
 		select {
 		case <-delayTimer.C:
-		case <-endCh:
-			return nil, errors.New("retry: timeout")
+			// Retry loop
```

Comment:
I feel like we're removing potentially valuable information here. Prior to this change the user would receive an error for timeout, _or_ an error for cancellation. With this change they'll no longer be able to distinguish between the two. Do we have good reason to no longer provide that information?

Reply:
Currently the behavior of this can be deceiving to the user. If `ctx.Done()` returns, it can either be due to cancellation or due to a time out. With either of those, our error returns `retry: cancelled`. There is an `Err()` function on context that we can use to differentiate between the two. I'll update the error message to include the context's error so they can differentiate.